File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
int main(void)
{
    float Change;
    int cents = 0, dollar = 0;
    do
    {
        Change = get_float("Change_owed: ");
    }
    while (Change < 0);
    if ((int) Change > 0)
    {
        dollar = ((int) Change);
        //number of cents (25)
        cents = (dollar * 100) / 25;
    }
    //the new change
    int New_change = round((Change - dollar) * 100);
    if (New_change != 0) //true
    {
        do //repeat
        {
       
            if (New_change >= 25)
            {
                //subtract 25 from change and add one to cents
                New_change -= 25; //true 
                cents++;
            }
            else if (New_change >= 10 && New_change < 25)
            {
                //subtract 10 from change and add one to cents
                New_change -= 10;
                cents++;

            }
            else if (New_change >= 5 && New_change < 10)
            {
                //subtract 5 from change and add one to cents
                New_change -= 5;
                cents++;

            }
            else if (New_change == 1)
            { 
                //subtract 5 from change and add one to cents
                cents++; 
                New_change--;
            }
        } // stop when change is 0
        while (New_change != 0);
    }
    printf("%i\n", cents);
    
    
}

File: cash.py
from cs50 import get_float

# ask the user for input b
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break
# converting dollars to cents
dollar = int(change) * 100
cents = dollar // 25
change = (change - int(change)) * 100

# number of cents
x1 = int(change) // 25
change = int(change) % 25
# number of 10
x2 = int(change) // 10
change = int(change) % 10
# number of 5
x3 = int(change) // 5
change = int(change) % 5
# number of 1
x4 = int(change) // 1
# print the total 
print(cents + x1 + x2 + x3 + x4)

